wpengine/atlas-content-modeler
# [MTKA-1451] Blueprints: support import/export of wp_options

## Description

Extends blueprint WP-CLI commands to support export and import of named WordPress options from the wp_options table.

No options are exported by default — developers must specify options they want to restore by passing the `--wp-options` flag during export..

https://wpengine.atlassian.net/browse/MTKA-1451

## Testing

Includes PHP unit tests.

To test option export manually:

1. Run `wp acm blueprint export --name="Options Test" --wp-options="blogname, blogdescription" --open`.
2. You should see a 'wp-options' key at the end of the generated `acm.json` file with your site's name and description.

To test option import manually:

1. Run `wp acm blueprint import http://contentmodel1.wpengine.com/wp-content/uploads/2022/03/options-test.zip`
3. Under Settings → General, your site name and tagline should have changed to “Custom Site Name” and “Custom site description”.

## Screenshots

```sh
> wp acm blueprint export --name="Options Test" --wp-options="blogname, blogdescription" --open
Collecting ACM data.
Collecting ACM models.
Collecting ACM taxonomies.
Collecting posts.
Collecting post tags.
Collecting post meta.
Collecting media.
Collecting ACM relationships
Writing acm.json manifest.
Generating zip.
Opening blueprint temp folder.
Success: Blueprint saved to /var/folders/mc/yf37tb4j5bjbwhv6z0m1bcfm0000gp/T/acm/options-test/options-test.zip.
```

```sh
> wp acm blueprint import http://contentmodel1.wpengine.com/wp-content/uploads/2022/03/options-test.zip
Fetching zip.
Unzipping.
Verifying ACM manifest.
Checking minimum versions.
Restoring WordPress options.
Deleting zip and manifest.
Success: Import complete.

> wp option get blogname
Custom Site Name

> wp option get blogdescription
Custom site description
```

## Documentation Changes

This PR includes docs updates.

## Dependant PRs

n/a
